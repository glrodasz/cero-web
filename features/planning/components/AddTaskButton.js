import PropTypes from 'prop-types'
import { Spacer, AddButton } from '@glrodasz/components'

import { MAXIMUM_BACKLOG_QUANTITY } from '../../../config'

const AddTaskButton = ({ id, tasksLength, onClickAddTask }) => {
  if (tasksLength < MAXIMUM_BACKLOG_QUANTITY) {
    return (
      <>
        <Spacer.Vertical size="md" />
        <AddButton
          id={id}
          onAdd={onClickAddTask}
          focusHelpText="Presiona enter"
          blurHelpText="Clic para continuar"
        >
          Toca para agregar la tarea
        </AddButton>
      </>
    )
  }

  return null
}

AddTaskButton.defaultProps = {
  id: '',
}

AddTaskButton.propTypes = {
  id: PropTypes.string,
  onClickAddTask: PropTypes.func.isRequired,
  tasksLength: PropTypes.number,
}

export default AddTaskButton
