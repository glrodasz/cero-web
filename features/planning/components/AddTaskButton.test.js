import AddTaskButton from './AddTaskButton'
import { render } from '@testing-library/react'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../utils/testUtils/dummyRender')
  return {
    Spacer: { Vertical: dummyRender('Spacer.Vertical') },
    AddButton: dummyRender('Button'),
  }
})

jest.mock('../../../config', () => ({
  MAXIMUM_BACKLOG_QUANTITY: 1,
}))

describe('[ features / planning / components / AddTaskButton ]', () => {
  describe('when `AddTaskButton` is mounted', () => {
    it('should render', () => {
      const props = {
        tasksLength: 0,
        onClickAddTask: () => {},
      }
      const { asFragment } = render(<AddTaskButton {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
