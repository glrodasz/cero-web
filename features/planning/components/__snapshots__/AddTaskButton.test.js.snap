// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / planning / components / AddTaskButton ] when \`AddTaskButton\` is mounted should render 1`] = `
<DocumentFragment>
  
[Spacer {"size":"md"} /]
[Button {"id":"","focusHelpText":"Presiona enter","blurHelpText":"Clic para continuar"}]Toca para agregar la tarea[/Button]
</DocumentFragment>
`;
